processor/processor.go

```diff
@@ -997,12 +985,22 @@ func (proc *HandleT) handlePendingGatewayJobs() bool {
 	proc.statDBR.Start()
 
 	toQuery := dbReadBatchSize
+	toProcessEvents := maxEventsToProcess
 	proc.logger.Debugf("Processor DB Read size: %v", toQuery)
 	//Should not have any failure while processing (in v0) so
 	//retryList should be empty. Remove the assert
-	retryList := proc.gatewayDB.GetToRetry([]string{gateway.CustomVal}, toQuery, nil)
-	toQuery -= len(retryList)
-	unprocessedList := proc.gatewayDB.GetUnprocessed([]string{gateway.CustomVal}, toQuery, nil)
+	unTruncatedRetryList := proc.gatewayDB.GetToRetry([]string{gateway.CustomVal}, toQuery, nil)
+	retryList, totalRetryEvents := getTruncatedEventList(unTruncatedRetryList, toProcessEvents)
+
+	toQuery -= totalRetryEvents
```

Comment:
Should this be 
toQuery -= len(retryList)
?

Reply:
dbReadBatchSize - 1000, maxLoopProcessEvents - 1000

assuming we got 500 events (200 requests) from querying retry jobs, it should be enough thatt we query for 500 unprocessed jobs rather than 800. (assuming 1 jobs has atleast one event)

Reply:
Got it. Looks good to me.

Given that we can assume there is at least one event in a job, if someone configures these variables wrong i.e., `dbReadSize > maxNumEvents in loop`

Should we make the

 `dbReadSize = min(dbReadSize, maxNumEvents)`

Reply:
Deciding `dbReadBatchSize` based on `avgEventsInRequest` is risky when we change `avgEventsInRequest` and restart. We may end up not processing events in order.